static/app/components/demoSandboxButton.tsx

```diff
@@ -0,0 +1,79 @@
+import Button from 'sentry/components/button';
+import {Organization, SandboxData} from 'sentry/types';
+import trackAdvancedAnalyticsEvent from 'sentry/utils/analytics/trackAdvancedAnalyticsEvent';
+import useOrganization from 'sentry/utils/useOrganization';
+
+type Props = {
+  /**
+   * The deep link scenario
+   */
+  scenario:
+    | 'performance'
+    | 'releases'
+    | 'alerts'
+    | 'discover'
+    | 'dashboards'
+    | 'projects'
+    | 'oneDiscoverQuery'
+    | 'oneIssue'
+    | 'oneBreadcrumb'
+    | 'oneStackTrace'
+    | 'oneTransaction'
+    | 'oneWebVitals'
+    | 'oneTransactionSummary'
+    | 'oneRelease';
+  /**
+   * Which project we should link to in the sandbox
+   */
+  projectSlug?: 'react' | 'python' | 'ios' | 'android' | 'react-native';
+  /**
+   * Matching on the error type or title
+   */
+  errorType?: string;
+
+  clientData?: SandboxData;
+} & React.ComponentProps<typeof Button>;
+
+/**
+ * Renders a button that will kick off the sandbox around children
+ * which should include be a button. If the sandbox is hidden,
+ * don't render the children
+ */
+function DemoSandboxButton({
```

Comment:
@Neo-Zhixing We already have this component in getsentry. Can we just add a hook in sentry instead of making a duplicate component? Long term, we probably should be moving as much of the sandbox functionality into hooks so the open-source product doesn't have as much sandbox logic.

Reply:
Long term we probably also want to show the sandbox button to self-serve customers, so this is probably not a getsentry only feature.

Reply:
@Neo-Zhixing Do you mean self-hosted customers?

Reply:
yeah my bad

Reply:
Decided to add this for self-hosted customers as well.